internal/command/clusters/uninstall.go

```diff
@@ -0,0 +1,218 @@
+			messagesIssuingCerts := []string{}
+			for _, cert := range certificates.Items {
+				issuing := false
+				ready := false
+				for _, cond := range cert.Status.Conditions {
+					if cond.Type == certmanagerv1.CertificateConditionIssuing && cond.Status == "True" {
+						issuing = true
+					}
+					if cond.Type == certmanagerv1.CertificateConditionReady && cond.Status == "True" {
+						ready = true
+					}
+				}
+				if issuing && !ready {
+					messagesIssuingCerts = append(messagesIssuingCerts, fmt.Sprintf("%s/%s", cert.Namespace, cert.Name))
+				}
+			}
+
+			// check for any potentially problematic certificate requests
+			messagesCertificateRequests := []string{}
+			certificateRequestClient, err := clients.NewCertificateRequestClient(kubeCfg)
+			if err != nil {
+				return fmt.Errorf("error creating certificate request client: %s", err)
+			}
+			var certificateRequests certmanagerv1.CertificateRequestList
+			err = certificateRequestClient.List(
+				ctx,
+				&clients.GenericRequestOptions{},
+				&certificateRequests,
+			)
+			if err != nil {
+				return fmt.Errorf("error listing certificate requests: %s", err)
+			}
+			for _, cr := range certificateRequests.Items {
+				ready, denied, approved := false, false, false
+				for _, cond := range cr.Status.Conditions {
+					if cond.Type == certmanagerv1.CertificateRequestConditionReady && cond.Status == "True" {
+						ready = true
+					}
+					if cond.Type == certmanagerv1.CertificateRequestConditionDenied && cond.Status == "True" {
+						denied = true
+					}
+					if cond.Type == certmanagerv1.CertificateRequestConditionApproved && cond.Status == "True" {
+						approved = true
+					}
+				}
+				if !approved && !denied {
+					messagesCertificateRequests = append(messagesCertificateRequests, fmt.Sprintf("%s/%s is pending approval", cr.Namespace, cr.Name))
+				}
+				if approved && !ready {
+					messagesCertificateRequests = append(messagesCertificateRequests, fmt.Sprintf("%s/%s is pending issuance", cr.Namespace, cr.Name))
+				}
+			}
+
+			// display information from the data gathered
+			messagesNextSteps := []string{}
+			if len(messagesOwnedSecrets) > 0 {
+				messagesNextSteps = append(messagesNextSteps, "Run 'jsctl experimental clusters cleanup secrets remove-certificate-owner-refs' to make sure secrets containing certificates are not garbage collected")
+				fmt.Fprintf(os.Stderr, "The following secrets contain certificates and are owned by a Certificate resource:\n")
+				for _, message := range messagesOwnedSecrets {
+					fmt.Fprintf(os.Stderr, " * %s\n", message)
+				}
+			}
+
+			if len(messagesUpcomingRenewals) > 0 {
+				cmctlRenewCmds := []string{}
+				for _, cert := range upcomingCerts {
+					cmctlRenewCmds = append(cmctlRenewCmds, fmt.Sprintf("cmctl renew --namespace=%s %s", cert[0], cert[1]))
+				}
+				messagesNextSteps = append(messagesNextSteps, fmt.Sprintf("Use cmctl to manually renew certificates: %s", strings.Join(cmctlRenewCmds, " && ")))
```

Comment:
Suggestion: do not tell the users to do manual renewal, in most cases it should not be needed. Instead we could:
- output cert's expiry time as well as renewal time
- print something like `make sure that you allocate enough time between uninstallation and re-installation to not cause any application downtime. If you would like to renew any of the certs now you can use cmctl...`